### PR TITLE
Feature/abort migration

### PR DESCRIPTION
**Pull request by begonaguereca**
Problem: Customer needed a way to delete a repo migration that was in progress was because of a bug or the migration is taking too long and they currently do not have a way to do this themselves. 

Solution: Adding a CLI abort migration command that allows users to self service 

Call outs: 

- Still need to [add documentation](https://github.ghe.com/github/octoshift/issues/7890)
- There is one edge case that is not covered, when a user is [aborting a migration that has already completed](https://github.ghe.com/github/octoshift/issues/7891). After some [discussion with some team member](https://github.slack.com/archives/C03R71V4HNK/p1702573948973709)s we have decided to proceed implementing this feature and adding this issue to the backlog. 

**ADO abort command:**

Add your creds if you haven't done that already: 
```
export GH_PAT="pat"
export ADO_PAT="pat"
```

Kick off a migration: 
```
dotnet run --project src/ado2gh/ado2gh.csproj -- migrate-repo --ado-org valet-testing --ado-team-project build-playground --ado-repo build-playground --github-org import-testing --github-repo abort-repo-testing
```

ABORT it: 
```
dotnet run --project src/ado2gh/ado2gh.csproj -- abort-migration --migration-id RM_kgDaACQyZjk0YzIxYy02ZWFjLTRmZjItYWM3Ni04ZWFkMTBkOWYxOWY
```
______________________________________________________________________________

**BBS abort command:** 

Add your creds if you haven't done that already: 
```
export BBS_USERNAME=bbsadmin
export GH_PAT="pat"
export BBS_PASSWORD=PASSWORD
```

Kick off a migration: 
```
dotnet run --project src/bbs2gh/bbs2gh.csproj -- migrate-repo --bbs-server-url https://test-bbs-o.githubapp.com/ --bbs-project IM --bbs-repo test-commit-id-blank-review
```

ABORT it: 
```
dotnet run --project src/bbs2gh/bbs2gh.csproj -- abort-migration --migration-id RM_kgDaACQyZjk0YzIxYy02ZWFjLTRmZjItYWM3Ni04ZWFkMTBkOWYxOWY
```

______________________________________________________________________________

**GEI abort command:** 

This is an example dotcom to dotcom migration. 

Add your creds if you haven't done that already: 
```
export GH_PAT="pat"
export GH_SOURCE_PAT="pat"
```

Kick off a migration: 
```
dotnet run --project src/gei/gei.csproj -- migrate-repo  --github-source-org valet-dev-testing --source-repo begona-test --github-target-org valet-testing-unit --target-repo new-repo-2
```

ABORT it: 
```
dotnet run --project src/gei/gei.csproj -- abort-migration --migration-id RM_kgDaACQyZjk0YzIxYy02ZWFjLTRmZjItYWM3Ni04ZWFkMTBkOWYxOWY
```

- [x] Did you write/update appropriate tests
- [ ] Release notes updated (if appropriate)
- [ ] Appropriate logging output
- [ ] Issue linked
- [ ] Docs updated (or issue created)
- [ ] New package licenses are added to `ThirdPartyNotices.txt` (if applicable)

Closes: https://github.ghe.com/github/octoshift/issues/7478

### CODE REVIEW

File: src/OctoshiftCLI.Tests/ado2gh/Commands/AbortMigration/AbortMigrationCommandTests.cs [hunk 1]
```diff
@@ -0,0 +1,21 @@
+using FluentAssertions;
+using OctoshiftCLI.AdoToGithub.Commands.WaitForMigration;
```
Comment (ArinGhazarian) [inline src/OctoshiftCLI.Tests/ado2gh/Commands/AbortMigration/AbortMigrationCommandTests.cs:2]: this is redundant
```suggestion
```

File: src/OctoshiftCLI.Tests/bbs2gh/Commands/AbortMigration/AbortMigrationCommandTests.cs [hunk 2]
```diff
@@ -0,0 +1,21 @@
+using FluentAssertions;
+using OctoshiftCLI.AdoToGithub.Commands.WaitForMigration;
```
Comment (ArinGhazarian) [inline src/OctoshiftCLI.Tests/bbs2gh/Commands/AbortMigration/AbortMigrationCommandTests.cs:2]: this is redundant
```suggestion
```

File: src/OctoshiftCLI.Tests/gei/Commands/AbortMigration/AbortMigrationCommandTests.cs [hunk 3]
```diff
@@ -0,0 +1,21 @@
+using FluentAssertions;
+using OctoshiftCLI.AdoToGithub.Commands.WaitForMigration;
```
Comment (ArinGhazarian) [inline src/OctoshiftCLI.Tests/gei/Commands/AbortMigration/AbortMigrationCommandTests.cs:2]:
```suggestion
```

**Review by ArinGhazarian — COMMENTED** [review]
Just a few nitpicks that I just noticed and releasenotes needs to be updated.

File: src/OctoshiftCLI.Tests/ado2gh/Commands/AbortMigration/AbortMigrationCommandTests.cs (hunk 1 above)
Comment (begonaguereca) [reply src/OctoshiftCLI.Tests/ado2gh/Commands/AbortMigration/AbortMigrationCommandTests.cs]: Shoot tests are broken, are you sure we dont need this ?

Comment (ArinGhazarian) [reply src/OctoshiftCLI.Tests/ado2gh/Commands/AbortMigration/AbortMigrationCommandTests.cs]: Oh the reason is the namespaces are wrong in `AbortMigrationCommand`s, will add new code suggestions.

File: src/OctoshiftCLI.Tests/gei/Commands/AbortMigration/AbortMigrationCommandTests.cs [hunk 4]
```diff
@@ -0,0 +1,20 @@
+using FluentAssertions;
```
Comment (ArinGhazarian) [inline src/OctoshiftCLI.Tests/gei/Commands/AbortMigration/AbortMigrationCommandTests.cs:1]:
```suggestion
using FluentAssertions;
using OctoshiftCLI.GithubEnterpriseImporter.Commands.AbortMigration;
```

File: releasenotes/v1.7.0.md [hunk 5]
```diff
@@ -0,0 +1,2 @@
+# Release Notes: 
+- Added `abort-migration` command that will abort a migration that are queued or in-progress.
```
Comment (ArinGhazarian) [inline releasenotes/v1.7.0.md:2]:
```suggestion
- Added `abort-migration` command to `gh ado2gh`, `gh bbs2gh` and `gh gei` that will abort a migration that is queued or in-progress. 
```

**Pull request merged**